guardian/mobile-n10n
# Increase lambda memory to the maximum

## What does this change?

Lambda can be configured with a maximum memory of 10240 MB. Increasing memory increases the CPU assigned to the lambda execution environment and has been [proven](https://github.com/guardian/mobile-n10n/blob/main/docs/testing/04-reducing-lambda-duration.md#memory) to reduce processing time as a result.

This change bumps the harvester and sender lambdas memory to 10240 MB so we can validate the performance improvement in production.

## How to test

After the change is deployed, check the aws console - the lambdas should show that 10240 MB memory has been configured.

## How can we measure success?

Monitor the 90in2 stat for improvements, we should also expect to see processing time decrease (we can use our [dashboard](https://eu-west-1.console.aws.amazon.com/cloudwatch/home?region=eu-west-1#dashboards:name=n10n-profiling) to monitor this).

## Have we considered potential risks?

There is an increased cost associated with bumping the lambda memory. The cost per invocation will increase by ~77%, however we can monitor the improvements and assess at a later date whether we believe this to be a worthwhile tradeoff.
